url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/com/dreamfabric/jac64/C64Reader.java

```diff
@@ -0,0 +1,756 @@
+/**
+ * This file is a part of JaC64 - a Java C64 Emulator
+ * Main Developer: Joakim Eriksson (Dreamfabric.com)
+ * Contact: joakime@sics.se
+ * Web: http://www.dreamfabric.com/c64
+ * ---------------------------------------------------
+ */
+
+package com.dreamfabric.jac64;
+/**
+ * @(#)Reader.java	Created date: 99-7-06
+ *
+ */
+import java.io.FileInputStream;
+import java.io.InputStream;
+import java.io.DataInputStream;
+import java.net.URL;
+import java.util.*;
+import java.io.OutputStream;
+import java.io.FileOutputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+
+/**
+ *
+ *
+ * @author  Joakim Eriksson (joakime@sics.se)
+ * @version $Revision: 1.5 $, $Date: 2006/05/01 14:57:57 $
+ */
+public class C64Reader {
+
+  public final static int NONE = 0;
+  public final static int TAPE = 1;
+  public final static int DISK = 2;
+
+  private int[] memory;
+  private String label = "";
+
+  private ArrayList dirNames = new ArrayList();
+  private Hashtable dirEntries = new Hashtable();
+
+  private DiskListener listener;
+
+  private int type = 0;
+  private int diskSize = 0; // Size in sectors
+
+  // This is typically a DISK but can also hold a tape!
+  private byte sectors[][] = new byte[800][256];
+
+  // For tape - counts the number of bytes stored in the
+  // sector array!!
+  private int noBytes;
+
+  public void setCPU(CPU cpu) {
+    /* CPU argument is ignored for now */
+    this.memory = cpu.getMemory();
+  }
+
+  public void setDiskListener(DiskListener l) {
+    listener = l;
+  }
+
+  public int getLoadedType() {
+    return type;
+  }
+  
+  public ArrayList getDirNames() {
+    return dirNames;
+  }
+
+  public DirEntry getDirEntry(String name) {
+    return (DirEntry) dirEntries.get(name);
+  }
+
+  public boolean readDisk(InputStream stream) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 17. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.